modules/timestamp/rewrite-fix-timezone.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2019 Balazs Scheidler <bazsi77@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#include "rewrite-fix-timezone.h"
+#include "scratch-buffers.h"
+#include "timeutils/cache.h"
+
+typedef struct _RewriteFixTimeZone RewriteFixTimeZone;
+
+struct _RewriteFixTimeZone
+{
+  LogRewrite super;
+  LogTemplate *zone_template;
+  gint stamp;
+};
+
+void
+rewrite_fix_time_zone_set_zone_template_ref(LogRewrite *s, LogTemplate *zone_template)
+{
+  RewriteFixTimeZone *self = (RewriteFixTimeZone *) s;
+
+  log_template_unref(self->zone_template);
+  self->zone_template = zone_template;
+}
+
+void
+rewrite_fix_time_zone_set_time_stamp(LogRewrite *s, gint stamp)
+{
+  RewriteFixTimeZone *self = (RewriteFixTimeZone *) s;
+  self->stamp = stamp;
+}
+
+static void
+_process(LogRewrite *s, LogMessage **pmsg, const LogPathOptions *path_options)
+{
+  RewriteFixTimeZone *self = (RewriteFixTimeZone *) s;
+  GString *result = scratch_buffers_alloc();
+  LogMessage *msg = *pmsg;
+
+  log_template_format(self->zone_template, msg, NULL, LTZ_LOCAL, 0, NULL, result);
```

Comment:
Instead of `LTZ_LOCAL`, `self->stamp` would be more suitable.

Reply:
That's a different beast. LTZ_LOCAL does not reference a specific timestamp (which self->stamp references) rather it selects which timezone the date macros would use: LTZ_LOCAL uses the timezone of the syslog-ng process (can be overrridden using the local-time-zone() setting in the options block)

LTZ_SEND would use the timezone setting of a destination. Since we are not even in an output context this one doesn't make sense.

Hope it is cleaner with this explanation.

Reply:
Ok I understand, thanks!